### ISSUE TITLE
How to define Cabal-like MIN_VERSION_base macros?

### ISSUE THREAD

**Issue by mrkkrp** [ndmitchell/hlint#178]
Hi, I'm using Flycheck (an Emacs package to check code on the fly), which in turn uses HLint among other tools. Currently I need to work with code where Cabal-defined macros are used, like this:

```haskell
#if MIN_VERSION_base(4,7,0)
import Data.Bool (bool)
#endif

…

#if !MIN_VERSION_base(4,8,0)
import Control.Applicative ((<$>), (<*))
#endif
#if !MIN_VERSION_base(4,7,0)
bool :: a -> a -> Bool -> a
bool f _ False = f
bool _ t True  = t
#endif
```

I'm already passing `-XCPP` argument and it enables C++ preprocessor, but now I'm getting `undefined MIN_VERSION_base macro` messages. So I guess I should use `--cpp-define` somehow. `--cpp-define=MIN_VERSION_base(a,b,c)=1` works, but I would like to define something like `--cpp-define=MIN_VERSION_base(a,b,c)=(a==4&&b==8&&c==0)` and that doesn't seem to work.

Flycheck also uses GHC, but I got away (seemingly) with just `-cpp` and `-DMIN_VERSION_base(a,b,c)` there.

**Comment by mrkkrp** [comment]
Ahh, nevermind. It seems GHC checker was failing, not HLint. What a hairy stuff. Great work, BTW.

**Comment by ndmitchell** [comment]
Hlint is pretty relaxed about cpp errors, so glad it wasn't causing issues.
